plumbing/transport/ssh/auth_method.go

```diff
@@ -311,13 +314,40 @@ type HostKeyCallbackHelper struct {
 	// HostKeyAlgorithms is a list of supported host key algorithms that will
 	// be used for host key verification.
 	HostKeyAlgorithms []string
+
+	// fallback allows for injecting the fallback call, which is called
+	// when a HostKeyCallback is not set.
+	fallback func(files ...string) (ssh.HostKeyCallback, error)
```

Comment:
Might be worth indicating that this is only meant for testing, but in practice it always calls `NewKnownHostsCallback`